main/src/ca/uwaterloo/flix/Main.scala

```diff
@@ -286,6 +286,13 @@ object Main {
     * @param args the arguments array.
     */
   def parseCmdOpts(args: Array[String]): Option[CmdOpts] = {
+    implicit val readInclusion: scopt.Read[LibLevel] = scopt.Read.reads {
```

Comment:
You would think that scopt could deal better with this? But maybe it cant?

Like do we really have to throw an exception?

Reply:
The exception seems to be the way scopt wants us to implement it, based on their implementations here: https://github.com/scopt/scopt/blob/develop/shared/src/main/scala/scopt/Read.scala

A `readsOpt` would be nice, but it's not available as far as I can tell.

Reply:
What error message does it print to the terminal then?

Reply:
```
Error: Option --Xlib failed when given 'blah'. illegal library argument
Try --help for more information.
Unable to parse command line arguments. Will now exit.
```

So I guess it takes the error message I provided. I'll make it more specific, like:
```
$badArg is not a valid library level. Valid options are "all", "min", "nix".
```